packages/tailwindcss/src/candidate.ts

```diff
@@ -523,21 +523,32 @@ function parseModifier(modifier: string): CandidateModifier | null {
   }
 
   if (modifier[0] === '(' && modifier[modifier.length - 1] === ')') {
-    let arbitraryValue = decodeArbitraryValue(modifier.slice(1, -1))
+    // Drop the `(` and `)` characters
+    modifier = modifier.slice(1, -1)
 
-    // Values can't contain `;` or `}` characters at the top-level.
-    if (!isValidArbitrary(arbitraryValue)) return null
+    // A modifier with `(…)` should always start with `--` since it
+    // represents a CSS variable.
+    if (modifier[0] !== '-' || modifier[1] !== '-') return null
 
-    // Empty arbitrary values are invalid. E.g.: `data-():`
-    //                                                 ^^
-    if (arbitraryValue.length === 0 || arbitraryValue.trim().length === 0) return null
+    // Trim the modifier to remove any leading or trailing whitespace
+    modifier = modifier.trim()
```

Comment:
Can this actually happen before decoding the value?

Reply:
You know, that is a good point actually!